### DISCUSSION TITLE
Passing values to python

### DISCUSSION THREAD

**Discussion by hseliger** [tndrle/PyLuaTeX#11] [Ideas]
Hi!

Firstly, thanks for this package! It's a great help!!

A little suggestion for an additional command: frequently I need to pass values from LaTeX to python to make handling easier, such as using package options from another package to set values used in the python code. This means TeX commands in the python code need to be expanded first. Currently I solve it in a simple manual way with an `\expandafter`, which would fail with more complex assigments. But maybe a starting point?

```
\documentclass{article}
\usepackage{pyluatex}
\begin{document}
	\def\foo{This is bar}
	\expandafter\pyc{a = "\foo"}
	\py{a}
\end{document}
```

Cheers,
Hendrik

**Comment by tndrle** [comment]
Hi Hendrik,

thank you for your suggestion. I often find myself using the same approach as yours (without the `\expandafter`, though).

Your example also works without `\expandafter`. Could you provide an example where you have problems, please?
To be honest, LaTeX expansion is a mystery to me and I cannot tell in which situations one would get problems here.

Tobias

**Reply by hseliger** [reply]
Hi Tobias,
you're right, this actually works without the expandafter. I just tired a few variations, but as long as the command insude the `\pyc` does not expand to another command, it works. I thought I had a python error message during writing of a little package that went away with the `\expandafter`, but I can't figure out anymore what that was. Maybe I had fixed something else at the same time and just believed the `\expandafter` did any good. Thanks for pointing this out!
– Hendrik
